### ISSUE TITLE
tsx下使用v-html报错

### ISSUE THREAD

**Issue by aoi-umi** [vuejs/vue#10307]
### Version
2.6.10

### Reproduction link
[https://github.com/aoi-umi/demo/commit/b738088f5d934b9c8e53a1f4a3e4a761966af7b0#diff-95d77e7cf3ad25bef47127a93ba4738b](https://github.com/aoi-umi/demo/commit/b738088f5d934b9c8e53a1f4a3e4a761966af7b0#diff-95d77e7cf3ad25bef47127a93ba4738b)

### Steps to reproduce
tsx下使用v-html
```html
<p v-html={params.row.content}></p>;
```
无法显示，报以下错误
Failed to resolve directive: html

### What is expected?
正确显示html内容

### What is actually happening?
Failed to resolve directive: html

**Comment by PreciousnessX** [comment]
<p v-html="params.row.content"></p>;

**Comment by PreciousnessX** [comment]
```<p v-html="params.row.content"></p>```

**Comment by aoi-umi** [comment]
> 
> 
> `<p v-html="params.row.content"></p>`

用tsx的，不是字符串模板

**Comment by sodatea** [comment]
https://github.com/vuejs/jsx#directives

```TSX
<p domPropsInnerHTML={html} />
```

TSX 和模板语法是不一样的

**Comment by dyf19118** [comment]
sorry for comment on closed issue, in vue 2.7.14 project, when using domPropsInnerHTML property on JSX elements in setup's return function, it raises error `Property 'domPropsInnerHTML' does not exist on type 'ElementAttrs<HTMLAttributes>'`, it seems that vue's jsx type definition only supports `innerHTML` but not `domPropsInnerHTML`

**Comment by dyf19118** [comment]
> https://github.com/vuejs/jsx#directives
> 
> ```tsx
> <p domPropsInnerHTML={html} />
> ```
> 
> TSX 和模板语法是不一样的

sorry for comment on closed issue, in vue 2.7.14 project, when using domPropsInnerHTML property on JSX elements in setup's return function, it raises error Property 'domPropsInnerHTML' does not exist on type 'ElementAttrs<HTMLAttributes>', it seems that vue's jsx type definition only supports innerHTML but not domPropsInnerHTML
